File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/pat/update/PersonalAccessTokenUpdate.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.update;

import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.id.IPersonalAccessTokenId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.pat.update.PersonalAccessTokenUpdate")
public class PersonalAccessTokenUpdate implements IUpdate, IObjectUpdate<IPersonalAccessTokenId>
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IPersonalAccessTokenId personalAccessTokenId;

    @JsonProperty
    private FieldUpdateValue<Date> accessDate = new FieldUpdateValue<Date>();

    @Override
    @JsonIgnore
    public IPersonalAccessTokenId getObjectId()
    {
        return getPersonalAccessTokenId();
    }

    @JsonIgnore
    public IPersonalAccessTokenId getPersonalAccessTokenId()
    {
        return personalAccessTokenId;
    }

    @JsonIgnore
    public void setPersonalAccessTokenId(IPersonalAccessTokenId personalAccessTokenId)
    {
        this.personalAccessTokenId = personalAccessTokenId;
    }

    @JsonIgnore
    public FieldUpdateValue<Date> getAccessDate()
    {
        return accessDate;
    }

    @JsonIgnore
    public void setAccessDate(Date accessDate)
    {
        this.accessDate.setValue(accessDate);
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("personalAccessTokenId", personalAccessTokenId).toString();
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/pat/update/PersonalAccessTokenUpdate.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/update/FieldUpdateValue" ], function(stjs, FieldUpdateValue) {
	var PersonalAccessTokenUpdate = function() {
		this.accessDate = new FieldUpdateValue();
	};
	stjs.extend(PersonalAccessTokenUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.pat.update.PersonalAccessTokenUpdate';
		constructor.serialVersionUID = 1;
		prototype.personalAccessTokenId = null;
		prototype.accessDate = null;

		prototype.getObjectId = function() {
			return this.getPersonalAccessTokenId();
		};
		prototype.getPersonalAccessTokenId = function() {
			return this.personalAccessTokenId;
		};
		prototype.setPersonalAccessTokenId = function(personalAccessTokenId) {
			this.personalAccessTokenId = personalAccessTokenId;
		};
		prototype.getAccessDate = function() {
			return this.accessDate;
		};
		prototype.setAccessDate = function(accessDate) {
			this.accessDate.setValue(accessDate);
		};
	}, {
		personalAccessTokenId : "IPersonalAccessTokenId",
		accessDate : {
			name : "FieldUpdateValue",
			arguments : [ "Date" ]
		}
	});
	return PersonalAccessTokenUpdate;
})